3D Chess/Assets/Scripts/Piece.cs

```diff
@@ -0,0 +1,28 @@
+using UnityEngine;
+
+public class Piece : MonoBehaviour
+{
+    public PieceType type = default;
+
+    private Events events = default;
+
+    public void Select()
+    {
+        gameObject.GetComponent<Renderer>().material.color = new Color(1, 0, 0, 1);
+    }
+
+    public void Deselect()
+    {
+        gameObject.GetComponent<Renderer>().material.color = new Color(1, 1, 1, 1);
+    }
+
+    private void OnMouseDown()
```

Comment:
Should this be `OnMouseUpAsButton` instead?

Reply:
https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnMouseUpAsButton.html only fires (in some cases quite literally) when the button is released. That's something we very rarely want in games. I'd guess that in 99.9% of cases a user would expect the action to happen when pressing, not when releasing (https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnMouseUpAsButton.html) the button.

Reply:
I would agree for real-time games, but with chess, I kind of imagine users to expect releasing the mouse to be the point at which a piece is selected or moved. I haven't run the game yet, just looking at the code, but I would expect that the piece wouldn't move until I release the mouse button, allowing me a few milliseconds to reconsider and cancel the move by dragging the pointer outside the square before releasing. That being said, this is a Realm demo, not necessarily a Unity one, so if our game isn't the most compelling game in the world, it's fine :D

Reply:
Interestingly enough a chess player would expect exactly the opposite. Especially in speed and blitz you want to move pieces as quickly as possible. To keep it consistent, chess games move on MouseDown not MouseUp (see for example chess.com to name just the biggest one out there).

That's just for the movement. You can still re-select another piece by clicking on it (also mirrored from chess.com).
Which - interestingly enough - is actually incorrect since according to the touch-move rule you have to move a piece after touching it (if there is a legal move for it).
Well, unless you say `j’adoube`, which is probably the first bit of French I ever learned thanks to Chess. :D
... apart from `en passant` (which is also not implemented in my version and must be done manually).

So, to be correct I would actually have to remove the option to select another piece (while keeping OnMouseDown) since reconsidering it not allowed.

Reply:
lol, okay, apparently my intuition about chess is completely off 😅

Reply:
Most of that I only know because I played competitively as a teen. Would have no clue otherwise. :D